homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -27,6 +27,15 @@ def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
             update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
         )
 
+    def get_client(self) -> bool:
+        """Connect to Yale API."""
+        self.yale = YaleSmartAlarmClient(
+            self.entry.data[CONF_USERNAME], self.entry.data[CONF_PASSWORD]
+        )
+        if self.yale:
```

Comment:
This will always be True. What is the problem we're trying to solve?

Reply:
This is to return a true/false to the setup e.g. faulty credentials would return a false to the setup and raise ConfigEntryAuthFailed https://github.com/home-assistant/core/pull/53577/files#diff-35b310117ded105edd0f0cd7283f08f0a2ed8d19c695124364d4090390930673R19

Reply:
Ok. How is the invalid authentication determined?

Reply:
With faulty credentials it will not be able to set self.yale and therefore return False to the setup.

Reply:
I don't think that's how it works. The library does a side effect and tries to authenticate against the api when instantiating the client. If it fails it will raise an error.

Where do we catch that error? Also, since the library doesn't support asyncio we need to run I/O in the executor. Finally, it's really bad pattern to do side effects when instantiating a class. I recommend to refactor the library in the future.

Reply:
You're right and I agree. The library was already there so some work has been done but still work to be done.
I will rework this in the code and it would also suit it better with a future improved version of the library.